Mathlib/AlgebraicGeometry/Curve/Smooth.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2024 Fangming Li. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fangming Li
+-/
+import Mathlib.AlgebraicGeometry.Scheme
+import Mathlib.RingTheory.Smooth.Basic
+
+/-!
+# Smooth morphisms of schemes
+
+In this file, we define when a morphism of schemes is smooth.
+-/
+
+/--
+Smooth homomorphisms between rings.
+-/
+def RingHom.IsSmooth {R A : Type _} [CommRing R] [CommRing A] (f : R →+* A) : Prop :=
+  @Algebra.Smooth R _ A _ (RingHom.toAlgebra f)
+
```

Comment:
This should be moved to `Mathlib.RingTheory.Smooth.Basic`. Could you please also add some basic API (i.e. compositions of smooth maps are smooth, etc.)? You can take a look at https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/FinitePresentation.html#RingHom.FinitePresentation as a reference.